lighthouse-core/test/audits/uses-rel-preload-test.js

```diff
@@ -9,56 +9,105 @@
 /* eslint-env mocha */
 
 const UsesRelPreload = require('../../audits/uses-rel-preload.js');
+const NetworkNode = require('../../lib/dependency-graph/network-node');
 const assert = require('assert');
 const defaultMainResource = {
   _endTime: 1,
 };
 
-const mockArtifacts = (networkRecords, mockChain, mainResource = defaultMainResource) => {
-  return {
-    devtoolsLogs: {
-      [UsesRelPreload.DEFAULT_PASS]: [],
-    },
-    requestCriticalRequestChains: () => {
-      return Promise.resolve(mockChain);
-    },
-    requestNetworkRecords: () => networkRecords,
-    requestMainResource: () => {
-      return Promise.resolve(mainResource);
-    },
+describe('Performance: uses-rel-preload audit', () => {
+  let mockGraph;
+  let mockSimulator;
+
+  const mockArtifacts = (networkRecords, mockChain, mainResource = defaultMainResource) => {
```

Comment:
the audit has some non-trivial uses of the graph and simulator, which means we're doing a lot of testing of the mocks :) Some tests using the full real thing would be 👍 👍

Reply:
totally agreed, but it requires a trace and devtoolslog of a site that would benefit from preloading which can be quite large.

my proposal:
* we rely on the smoke test + unit tests for now
* for cases like this where we need to exercise complicated lantern paths with real traces we use my patent-pending post-I/O approach of storing ~100 traces/devtoolslogs on GCP that can run on-demand/on commits that touch lantern files to assert that opportunities and metrics look decent

WDYT?

Reply:
it wouldn't give good coverage, but even a simple devtoolslog/trace with little or no savings would be a start until the GCP smoke tests. there's `lighthouse-core/tests/fixtures/artifacts/perflog/` and of course `lighthouse-core/test/results/artifacts/`

Reply:
sure

Reply:
we should maybe make a tool to take real artifacts and strip out a bunch of stuff to make nice artifact fixtures and maybe only a few hundred KB at most. e.g. for things like this that are mostly networking we can get rid of a lot of trace events (enough that it was the same page load but with a lot less busy CPU)

(we have a lot of old traces for testing, but without paired devtools logs they're becoming increasingly useless for tests)